azure-iot-device/samples/advanced-hub-scenarios/send_to_output_x509.py

```diff
@@ -0,0 +1,64 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import os
+import uuid
+from azure.iot.device.aio import IoTHubModuleClient
+from azure.iot.device.iothub import Message
+from azure.iot.device.common import X509
+import logging
+import asyncio
+
+
+logging.basicConfig(level=logging.DEBUG)
+
+messages_to_send = 10
+
+
+async def main():
+    hostname = os.getenv("HOSTNAME")
+
+    # The device having a certain module that has been created on the portal
+    # using X509 CA signing or Self signing capabilities
+
+    device_id = os.getenv("DEVICE_ID")
+    module_id = os.getenv("MODULE_ID")
+
+    x509 = X509(
+        cert_file=os.getenv("X509_CERT_FILE"),
+        key_file=os.getenv("X509_KEY_FILE"),
+        pass_phrase=os.getenv("PASS_PHRASE"),
+    )
+
+    module_client = IoTHubModuleClient.create_from_x509_certificate(
+        hostname=hostname, x509=x509, device_id=device_id, module_id=module_id
+    )
+
+    # Connect the client.
+    await module_client.connect()
+
+    async def send_test_message(i):
+        print("sending message #" + str(i))
+        msg = Message("test wind speed " + str(i))
+        msg.message_id = uuid.uuid4()
+        msg.correlation_id = "correlation-1234"
+        msg.custom_properties["tornado-warning"] = "yes"
+        await module_client.send_to_output(msg, "twister")
```

Comment:
send_to_output is only "officially supported" for iot edge modules, but iot edge doesn't (as far as I know) support x509 auth, so the premise of this sample may be invalid.

I say not "officially supported" because you _can_ use send_to_output with iot hub, but it arrives at iothub as telemetry with some extra attributes.  (At least you could do this at one point).  This is not documented or supported behavior AFAIK.